src/components/MAPI/apps/__tests__/ClusterDetailAppListWidgetUninstall.tsx

```diff
@@ -0,0 +1,183 @@
+
+function getComponent(
+  props: React.ComponentPropsWithoutRef<
+    typeof ClusterDetailAppListWidgetUninstall
+  >
+) {
+  const history = createMemoryHistory();
+  const auth = new TestOAuth2(history, true);
+
+  const Component = (p: typeof props) => (
+    <ClusterDetailAppListWidgetUninstall {...p} />
+  );
+
+  return getComponentWithStore(
+    Component,
+    props,
+    undefined,
+    undefined,
+    history,
+    auth
+  );
+}
+
+function generateApp(
+  name: string = 'some-app',
+  namespace: string = 'j5y9m'
+): applicationv1alpha1.IApp {
+  return {
+    apiVersion: 'application.giantswarm.io/v1alpha1',
+    kind: 'App',
+    metadata: {
+      annotations: {
+        'chart-operator.giantswarm.io/force-helm-upgrade': 'true',
+      },
+      creationTimestamp: new Date().toISOString(),
+      finalizers: ['operatorkit.giantswarm.io/app-operator-app'],
+      generation: 1,
+      labels: {
+        app: name,
+        'app-operator.giantswarm.io/version': '3.2.1',
+        'giantswarm.io/cluster': namespace,
+        'giantswarm.io/managed-by': 'Helm',
+        'giantswarm.io/organization': 'org1',
+        'giantswarm.io/service-type': 'managed',
+      },
+      name,
+      namespace,
+      resourceVersion: '294675096',
+      selfLink: `/apis/application.giantswarm.io/v1alpha1/namespaces/${namespace}/apps/${name}`,
+      uid: '859c4eb1-ece4-4eca-85b2-a4a456b6ae81',
+    },
+    spec: {
+      catalog: 'default',
+      config: {
+        configMap: {
+          name: `${namespace}-cluster-values`,
+          namespace,
+        },
+        secret: {
+          name: '',
+          namespace: '',
+        },
+      },
+      kubeConfig: {
+        context: {
+          name: `${namespace}-kubeconfig`,
+        },
+        inCluster: false,
+        secret: {
+          name: `${namespace}-kubeconfig`,
+          namespace,
+        },
+      },
+      name,
+      namespace: 'giantswarm',
+      userConfig: {
+        configMap: {
+          name: '',
+          namespace: '',
+        },
+        secret: {
+          name: '',
+          namespace: '',
+        },
+      },
+      version: '0.4.1',
+    },
+    status: {
+      appVersion: '0.4.1',
+      release: {
+        lastDeployed: '2021-04-27T16:21:37Z',
+        status,
+      },
+      version: '0.4.1',
+    },
+  };
+}
+
+describe('ClusterDetailAppWidgetUninstall', () => {
+  it('renders without crashing', () => {
+    render(getComponent({}));
+  });
+
+  it('displays the button to uninstall an app', () => {
+    const app = generateApp();
+    render(getComponent({ app }));
+    expect(screen.getByText('Uninstall...')).toBeInTheDocument();
+  });
+
+  it('displays the confirmation for uninstalling an app when the uninstall button is clicked', () => {
+    const app = generateApp();
+
+    render(getComponent({ app }));
+
+    const uninstallButton = screen.getByRole('button', {
+      name: 'Uninstall...',
+    }) as HTMLButtonElement;
```

Comment:
Do we need this assertion?

```suggestion
    });
```